alws/crud/test.py

```diff
@@ -141,3 +147,66 @@ async def get_test_tasks_by_build_task(
             query = query.filter(models.TestTask.revision == revision)
         result = await db.execute(query)
         return result.scalars().all()
+
+
+async def get_test_logs(build_id: int, db: Session) -> dict:
+    """
+    Parses test logs and determine test format.
+    Returns dict of lists for each build's test with detailed status report.
+
+    Parameters
+    ----------
+    build_id : int
+        Id of the build, which tests info is interested
+    db: Session
+        db connection
+
+    Returns
+    -------
+    dict
+
+    """
+    async with db.begin():
+        test_task_query = select(models.TestTask).where(
+            models.TestTask.build_task_id == build_id)
+        result = await db.execute(test_task_query)
+        test_task = result.scalars().all()
+        repo_id = test_task[0].repository_id
```

Comment:
Maybe there is a point to avoid using index with `result.scalars().first()`
Also there is a point to rename `test_task` on `test_tasks`